pkg/controller/scansettingbinding/scansettingbinding_controller.go

```diff
@@ -0,0 +1,496 @@
+	constraint := unstructured.Unstructured{}
+
+	constraint.SetAPIVersion(constraintRef.APIGroup)
+	constraint.SetKind(constraintRef.Kind)
+	logger.Info("Looking for constraint", "object", constraint)
+
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: constraintRef.Name}, &constraint); err != nil {
+		logger.Error(err, "Cannot find NamedObjectReference", "Setting.Name", constraintRef.Name)
+		return err
+	}
+	logger.Info("constraint object details", "kind", constraint.GetKind(), "apiVersion", constraint.GetAPIVersion())
+
+	expGvk := schema.GroupVersionKind{
+		Group:   compliancev1alpha1.SchemeGroupVersion.Group,
+		Version: compliancev1alpha1.SchemeGroupVersion.Version,
+		Kind:    "ScanSetting",
+	}
+
+	if err := isGvk(&constraint, &expGvk); err != nil {
+		return err
+	}
+	v1setting := compliancev1alpha1.ScanSetting{}
+
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(constraint.Object, &v1setting); err != nil {
+		return err
+	}
+
+	suite.Spec.AutoApplyRemediations = v1setting.AutoApplyRemediations
+	suite.Spec.Schedule = v1setting.Schedule
+
+	scansWithSelector := make([]compliancev1alpha1.ComplianceScanSpecWrapper, 0)
+	for _, scan := range suite.Spec.Scans {
+		logger.Info("Processing original scan", "scan.Name", scan.Name)
+		if scan.ScanType == compliancev1alpha1.ScanTypeNode {
+			for _, role := range v1setting.Roles {
+				scanCopy := scan.DeepCopy()
+				scanCopy.Debug = v1setting.Debug
+				scanCopy.Name = scan.Name + "-" + role
+				scanCopy.NodeSelector = utils.GetNodeRoleSelector(role)
+				logger.Info("Adding per-role scan", "scanCopy.Name", scanCopy.Name)
+				scansWithSelector = append(scansWithSelector, *scanCopy)
+			}
+		} else {
+			scanCopy := scan.DeepCopy()
+			scanCopy.Debug = v1setting.Debug
+			logger.Info("Adding platform scan", "scanCopy.Name", scanCopy.Name)
+			scansWithSelector = append(scansWithSelector, *scanCopy)
+		}
+
+	}
+
+	suite.Spec.Scans = scansWithSelector
+	return nil
+}
+
+func newCompScanFromBindingProfile(r *ReconcileScanSettingBinding, instance *compliancev1alpha1.ScanSettingBinding, ss *compliancev1alpha1.NamedObjectReference, logger logr.Logger) (*compliancev1alpha1.ComplianceScanSpecWrapper, string, error) {
+	parsedProfReference, err := resolveProfileReference(r, instance, ss, logger)
+	if err != nil {
+		return nil, "", err
+	}
+
+	scan, platform, err := profileReferenceToScan(parsedProfReference)
+	if err != nil {
+		if r.recorder != nil {
+			r.recorder.Eventf(
+				instance, corev1.EventTypeWarning, "ScanCreateError",
+				"Cannot create scan: %v", err,
+			)
+		}
+		return nil, "", err
+	}
+
+	return scan, platform, nil
+}
+
+type profileReference struct {
+	name string
+
+	tailoredProfile *unstructured.Unstructured
+	profile         *unstructured.Unstructured
+	profileBundle   *unstructured.Unstructured
```

Comment:
is there a reason why you're using unstructured objects here as opposed to the types themselves?

Reply:
Hopefully this is answered above.